cmake/modules/RootBuildOptions.cmake

```diff
@@ -406,6 +404,15 @@ foreach(opt )
   endif()
 endforeach()
 
+if(DEFINED rpath)
+  message(DEPRECATION ">>> Option 'rpath' is deprecated and is ignored."
+      " Relative RPATHs to the main ROOT libraries are unconditionally appended to all ROOT"
+      " executables and libraries. The 'rpath' flag presence will result in a configuration"
+      " error in the next release of ROOT."
+      " Please contact root-dev@cern.ch should you still need this option."
```

Comment:
This could prompt people to say "I need rpath", even if that's a misunderstanding. Therefore, I would link to the release notes and try to explain in there that they *still* are getting rpaths.